### PR TITLE
switch swift-tokenizers to main, remove some workarounds

### PR DESCRIPTION
**Pull request by davidkoski**
- swift-tokenizers is getting a lot of updates and fixes, let's track main for now
- remove some workarounds that are no longer needed

- https://github.com/huggingface/swift-transformers/issues/63

### CODE REVIEW

File: Libraries/LLM/Tokenizer.swift [hunk 1]
```diff
@@ -67,54 +67,13 @@ public func loadTokenizer(configuration: ModelConfiguration) async throws -> Tok
...
 
-    // workaround: some merges can't be split on space in BPETokenizer
-    if let tokenizerClass = tokenizerConfig.tokenizerClass?.stringValue {
-        switch tokenizerClass {
-        case "T5Tokenizer":
-            break
-        default:
-            tokenizerData = discardUnhandledMerges(tokenizerData: tokenizerData)
-        }
-    }
```
Comment (davidkoski) [inline Libraries/LLM/Tokenizer.swift:78]: This is fixed in:

- https://github.com/huggingface/swift-transformers/issues/51
- https://github.com/huggingface/swift-transformers/pull/52

Removing workaround

File: Libraries/LLM/Tokenizer.swift [hunk 2]
```diff
@@ -67,54 +67,13 @@ public func loadTokenizer(configuration: ModelConfiguration) async throws -> Tok
...
-
-    return tokenizerData
-}
-
 /// overrides for TokenizerModel/knownTokenizers
 let replacementTokenizers = [
-    "CodeLlamaTokenizer": "LlamaTokenizer",
-    "GemmaTokenizer": "PreTrainedTokenizer",
```
Comment (davidkoski) [inline Libraries/LLM/Tokenizer.swift:118]: These are now entries in knownTokenizers:

- https://github.com/huggingface/swift-transformers/blob/24605a8c0cc974bec5b94a6752eb687bae77db31/Sources/Tokenizers/Tokenizer.swift#L62

File: mlx-swift-examples.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 3]
```diff
@@ -59,8 +59,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/huggingface/swift-transformers",
       "state" : {
-        "revision" : "564442fba36b0b694d730a62d0593e5f54043b55",
-        "version" : "0.1.2"
+        "branch" : "main",
+        "revision" : "24605a8c0cc974bec5b94a6752eb687bae77db31"
```
[the remark this answers is not in the document]
Comment (pcuenca) [reply mlx-swift-examples.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:63]: I just pushed tag `0.1.3` in case it helps.

**Review by awni — APPROVED** [review]
❤️  all that red, thanks!

PS thanks @pcuenca for the quick turn-around fixes on Swift Transformers!

**Pull request merged**